### ISSUE TITLE
Are these metrics normal? Visdrone YOLO11n 

### ISSUE THREAD

**Issue by mldl4** [ultralytics/ultralytics#18031]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi! First of all, thank you for the project and for the activity in the repository. It is appreciated.
Right now, I am working with YOLO11n and Visdrone dataset and I'm having problems during training.
My model doesn't learn as I expected. Nevertheless, not good results.
Several tries with different hyperparameters have been done: 
- Default training 
- https://github.com/ultralytics/ultralytics/issues/17536
- https://github.com/ultralytics/ultralytics/issues/14101 

There is something I am doing wrong? or Should I perform some data augmentation or balancing process to obtain better results?

Thank you in advance!!!

PD: Next, I show my setup, libs and code.

# Setup

```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.06             Driver Version: 535.183.06   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce GTX 1080 Ti     On  | 00000000:01:00.0 Off |                  N/A |
|  0%   44C    P8              26W / 320W |      9MiB / 11264MiB |      0%      Default |
|                                         |                      |                  N/A |
|=========================================+======================+======================|

```

```
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Byte Order:                           Little Endian
Address sizes:                        46 bits physical, 48 bits virtual
CPU(s):                               24
On-line CPU(s) list:                  0-23
Thread(s) per core:                   1
Core(s) per socket:                   16
Socket(s):                            1
NUMA node(s):                         1
Vendor ID:                            GenuineIntel
CPU family:                           6
Model:                                151
Model name:                           12th Gen Intel(R) Core(TM) i9-12900K
```

```
20.04.1-Ubuntu
```

# Code

```
from datetime import datetime
from ultralytics import YOLO
import torch
import yaml
import json
import gc

torch.cuda.empty_cache()
gc.collect()


yaml_params = {}
with open('parameters.yaml', 'r') as yaml_file:
    yaml_params = yaml.safe_load(yaml_file)

model = YOLO("yolo11n.yaml").load("yolo11n.pt")
model.train(**params)

params_val = {
    'save_json': True,
    'project': 'eval' + params['name']
}
metrics = model.val(**params_val) 

```

# Hyperparameters

```
custom:
  data: visdrone.yaml
  patience: 15
  epochs: 300
  imgsz: 640 # 928
  batch: 8
  save: True
  device: 0
  plots: True
  seed: 42
  optimizer: AdamW
  dropout: 0.1
  lr0: 0.0007856857999453613
  augment: True
  multi_scale: True
  close_mosaic: 10
  workers: 8
  save_period: 1
  verbose: True
  # StepLR: True
  # gamma: 0.1
  # step_size: 50
  weight_decay: 0.000130
  pretrained: True
  val: True
  mixup: 0.1948
  momentum: 0.92551
  # cutmix: 0.3155
```

# Results
![image](https://github.com/user-attachments/assets/841e5217-13cf-47ba-a53e-9024b1da56c8)

## Training

![image](https://github.com/user-attachments/assets/8864f31b-604a-44b4-8170-c4a4fa581787)

## Eval
![image](https://github.com/user-attachments/assets/f34ef691-2a52-4b26-9ebb-0a53db278989)

```
mAP50-95 -> 0.1858984065721675
mAP50 -> 0.31818722095404556
mAP75 -> 0.18746636795979413

```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mldl4, thank you for your interest in Ultralytics 🚀! We appreciate the time you’ve taken to provide detailed information about your setup and experiments. This makes it much easier to evaluate potential issues.

If this is a 🐛 Bug Report or unexpected behavior, please provide a minimum reproducible example to help us debug it. An MRE should ideally include a small dataset (if possible) and the exact training or evaluation commands/code resulting in the issue. This allows us to reproduce the problem in a controlled environment.

If this is a ❓ Question regarding custom training and hyperparameters, it seems you've already tried various setups. It would help to provide additional information if possible, such as:
- Clarification on what specific challenges you've faced or what doesn't meet your expectations (e.g., is it the mAP50, the overall training trend, etc.).
- More details about your dataset (e.g., number of images per class, distribution, or any potential imbalance).
- Logs showing what happens during training and evaluation.

To ensure you're working with the latest features and fixes, we also recommend ensuring you're using the latest version of the `ultralytics` package and that your environment is fully up to date with the required dependencies.

Lastly, if you suspect this could be related to dataset-specific challenges, data augmentation, or preprocessing might indeed help. You could experiment with advanced augmentation techniques, ensuring data balance, or even tweaking learning parameters like `lr0`, `momentum`, and other model settings.

Rest assured, this is an automated response 🤖, but an Ultralytics engineer will assist you as soon as possible. In the meantime, thank you for your patience and for being part of our community! 🚀

**Comment by Y-T-G** [comment]
VisDrone has small objects, so you should probably used a larger imgsz. YOLO-OBB trained on DOTA for example uses imgsz=1024.

**Comment by mldl4** [comment]
Thank U, at this moment is working!!

**Comment by glenn-jocher** [comment]
@mldl4 you're welcome! Glad to hear it's working now. If you have any further questions, feel free to ask.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in PR #19870, which introduces the new CutMix image augmentation feature in Ultralytics 8.3.119. With CutMix, users can now apply an advanced augmentation technique that improves model robustness by simulating realistic occlusions and challenging scenarios during training. This update adds a cutmix option alongside existing methods like mosaic and mixup, updates configuration files, and enhances the documentation with detailed guides and visuals. The integration is seamless, allowing flexible tuning and easy adoption within the training pipeline.

Special thanks to the PR author and contributors for their efforts in bringing this feature to Ultralytics.

To test if this resolves your issue, please update your package with:
```
pip install -U ultralytics>=8.3.119
```
Alternatively, you can install the latest code from the main branch:
```
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Feel free to wait for the next official PyPI release if you prefer.

Your feedback is appreciated—please let us know if this update resolves the problem or if you encounter any further issues. Thank you for reporting this and helping improve Ultralytics! 🙏

**Comment by glenn-jocher** [comment]
Great to hear that increasing the image size helped with your VisDrone training! This dataset is particularly challenging due to its small objects, which often benefit from larger input resolutions. For reference, YOLO models trained on similar aerial imagery datasets (like DOTA) often use imgsz=1024 or higher to capture those details effectively.

You might also want to try the new CutMix augmentation feature mentioned by the assistant (available in v8.3.119+), which can further improve model robustness for datasets with small objects and occlusions. Combined with your existing mixup and multi-scale settings, it could potentially enhance performance even more.
